extensions/applicationinsights-dependencies-js/src/ajax.ts

```diff
@@ -334,59 +334,76 @@ export class AjaxMonitor extends BaseTelemetryPlugin implements IDependenciesPlu
+                                init = _self.includeCorrelationHeaders(fetchData, input, init);
+                                ctx.data = fetchData;
+                            }
+                        },
+                        rsp: (callDetails: IInstrumentCallDetails, input) => {
+                            let fetchData = callDetails.ctx().data;
+                            if (fetchData) {
+                                // Replace the result with the new promise from this code
+                                callDetails.rslt = callDetails.rslt.then(response => {
+                                    _reportFetchMetrics(callDetails, (response||{}).status, response, fetchData, () => {
+                                        let ajaxResponse:IAjaxRecordResponse = {
+                                            statusText: response.statusText,
+                                            headerMap: null,
+                                            correlationContext: _getFetchCorrelationContext(response)
+                                        };
+                    
+                                        if (_enableResponseHeaderTracking) {
+                                            const responseHeaderMap = {};
+                                            response.headers.forEach((value, name) => {
+                                                responseHeaderMap[name] = value;
+                                            });
+                    
+                                            ajaxResponse.headerMap = responseHeaderMap;
+                                        }
+                    
+                                        return ajaxResponse;
+                                    });
+                    
+                                    return response;
+                                })
+                                .catch(reason => {
+                                    _reportFetchMetrics(callDetails, 0, input, fetchData, null, { error: reason.message });
+                                    throw reason;
                                 });
-                
-                                return response;
-                            })
-                            .catch(reason => {
-                                _reportFetchMetrics(callDetails, 0, input, fetchData, null, { error: reason.message });
-                                throw reason;
-                            });
+                            }
+                        },
+                        // Create an error callback to report any hook errors
+                        hkErr: _createErrorCallbackFunc(_self, _InternalMessageId.FailedMonitorAjaxOpen, 
+                            "Failed to monitor Window.fetch, monitoring data for this fetch call may be incorrect.")
+                    }));
+    
+                    _fetchInitialized = true;
+                } else if ((fetch as any).polyfill) {
```

Comment:
Is it possible to this to be called even if explicitly configured off using disableFetchTracking?

Reply:
Yes, I removed the explict check from the constructor and added it above.
And this is needed to ensure that we capture this state.